### ISSUE TITLE
random_cycle tests still failing on the microcontroller

### ISSUE THREAD

**Issue by OpenBagTwo** [OpenBagTwo/HelloNeoPixel#2]
## Description

Running through the test suite on my ESP32 proto-board is showing an intermittent test failure in

`test_colors_are_shifted_by_one_pixel_after_transition_time (TestRandomCycle)`
https://github.com/OpenBagTwo/HelloNeoPixel/blob/d9c02f47df2e1ff68be12190f2c356e7c1a8227a/hello_neopixel/tests/test_animations.py#L127-L149

This is _after_ that test was made easier in 4faa7762e258bb39ed227abf60d1c000cd37e1fd

### Platforms and Versions
- MicroPython v1.14 on 2021-02-02; ESP32 module (spiram) with ESP32
- `hello_neopixel` v2021.3.8+4.gd9c02f4

## To Reproduce

```python
>>> import unittest
>>> unittest.main("hello_neopixel.tests")
```

(note: [requires patched version of `unittest` module](https://github.com/micropython/micropython-lib/issues/414))

## Expected Behavior
All tests should reliably pass.

## Workaround
Test sometimes passes

## Desired Resolution
It occurs to me that this isn't so much a fault of the test as it is a fault of the animation implementation. To wit: if I'm wanting to hit a certain number of frames per second, and I know that computation takes time, my sleep time shouldn't be 1 second / FPS.

**Comment by OpenBagTwo** [comment]
So my one concern with adjusting for compute-time was that I wasn't sure how costly it was for the microcontroller to poll the system clock. Turns out that concern was ludicrous:

```python
"""Timing how expensive it is to poll the system clock. Commented test results
were obtained using:
  - MicroPython v1.14 on 2021-02-02; ESP32 module (spiram) with ESP32"""
import utime

print("Method 1")
# concise and, perhaps, optimized?
ticks = [utime.ticks_ms() for _ in range(20)]
diffs = [utime.ticks_diff(ticks[i + 1], ticks[i]) for i in range(19)]
print(diffs)
print(utime.ticks_diff(ticks[-1], ticks[1]))
# yielded
# [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
# 0

print("Method 2")
# okay, at least force Python to do it as individual interpreter lines
ticks = []
for _ in range(20):
    ticks.append(utime.ticks_ms())
diffs = [ticks[i + 1] - ticks[i] for i in range(19)]
print(diffs)
print(utime.ticks_diff(ticks[-1], ticks[1]))
# yielded
# [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]
# 1

print("Method 3")
# let's time ticks_diff as well
diffs = []
last_tick = None
for _ in range(20):
    tick = utime.ticks_ms()
    if last_tick:
        diffs.append(utime.ticks_diff(tick, last_tick))
    else:
        first_tick = tick
    last_tick = tick
print(diffs)
print(utime.ticks_diff(last_tick, first_tick))
# yielded
# [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0]
# 2

print("Method 3b")
# let's go to microseconds!
diffs = []
last_tick = None
for _ in range(20):
    tick = utime.ticks_us()
    if last_tick:
        diffs.append(utime.ticks_diff(tick, last_tick))
    else:
        first_tick = tick
    last_tick = tick
print(diffs)
print(utime.ticks_diff(last_tick, first_tick))
# yielded
# [37, 119, 91, 108, 96, 93, 91, 92, 92, 98, 91, 92, 91, 99, 107, 96, 91, 101, 92]
# 1777

print("Final Test")
# Ludicrous mode
reference = utime.ticks_us()
start = utime.time_ns()
_ = [utime.ticks_diff(utime.ticks_us(), reference) for _ in range(1000)]
print(
    "1000 calls of ticks_diff + ticks_us took {:.6g} seconds".format(
        (utime.time_ns() - start) / 1e9
    )
)
# yielded
# 1000 calls of ticks_diff + ticks_us took 0.049304 seconds
```

So ~50μs* per combination of `ticks_us` and `ticks_diff`. That is _beyond_ a non-issue.

*max. There's memory allocation, variable assignment and string formatting(?) going on in this timespan as well.

**Comment by OpenBagTwo** [comment]
Also worthy of note:

```python
>>> utime.ticks_us() / utime.ticks_ms()
1000.001
```

That is, it seems likely that both flavors of ticks start at the same zero, meaning I can use microsecond-timing for some things without having to worry about running out of ticks in [1.2 hours](https://www.google.com/search?q=2^32+microseconds+to+hours)

**Comment by OpenBagTwo** [comment]
Resolved by c94bbcd216d6ff340e9659af1937591181142f58
